elastic/kibana
# [Infra] Replace lens attribute builder

closes [#173720](https://github.com/elastic/kibana/issues/173720)

## Summary

This PR replaces the usage of the Lens attribute builder utils created by the Obs UX Infra Services team, in favor of the solution created and maintained by the Visualizations team.

Hosts View

<img width="477" alt="image" src="https://github.com/elastic/kibana/assets/2767137/401029da-b8f6-4a16-869e-3c83a7823c26">
<img width="477" alt="image" src="https://github.com/elastic/kibana/assets/2767137/debbc763-fb3c-4e2d-b90a-9bb85df227d9">
<img width="477" alt="image" src="https://github.com/elastic/kibana/assets/2767137/28bbf07f-d9d2-4100-a84d-44d24bf35573">

Asset Details

<img width="477" alt="image" src="https://github.com/elastic/kibana/assets/2767137/13218f5a-ec9c-40d8-8348-b980172b4fed">
<img width="477" alt="image" src="https://github.com/elastic/kibana/assets/2767137/4c650f08-3d93-427a-a5c5-88b607942ce1">
<img width="478" alt="image" src="https://github.com/elastic/kibana/assets/2767137/868de5ec-5a67-46b7-8cdc-b49373267a7b">


Besides that, a few things in the charts were fixed:
- Centralization of the loading indicator
- Return of the bounds (0% to 100%) for line charts with percentage values (except for  Normalized Load)
- Return of the reference line for the Normalized Load chart.

### For reviewers

- Most of the changes are in the metrics-data-access plugin, where the chart and formula objects were adapted to the new interface. I've also grouped the formulas by metric group instead of having 1 file per formula.


## How to test
- setup a local Kibana instance
- Navigate to Infrastructure > Hosts
- Check if the charts render properly (Do the same for the Asset Details page and Asset Details flyout)
  - Test if the charts correctly redirects to Lens.
